pkg/agent/orm/topology/manager.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+import (
+	"fmt"
+	"sync"
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// maxAllowableNUMANodes specifies the maximum number of NUMA Nodes that
+	// the TopologyManager supports on the underlying machine.
+	//
+	// At present, having more than this number of NUMA Nodes will result in a
+	// state explosion when trying to enumerate possible NUMAAffinity masks and
+	// generate hints for them. As such, if more NUMA Nodes than this are
+	// present on a machine and the TopologyManager is enabled, an error will
+	// be returned and the TopologyManager will not be loaded.
+	maxAllowableNUMANodes = 8
+	// defaultResourceKey is the key to store the default hint for those resourceNames
+	// which don't specify hint.
+	defaultResourceKey = "*"
+)
+
+type Manager interface {
+	Admit(pod *v1.Pod) error
+
+	AddHintProvider(provider HintProvider)
+
+	GetAffinity(podUID string, containerName string, resourceName string) TopologyHint
+
+	RemovePod(podUID string)
+}
+
+// HintProvider is an interface for components that want to collaborate to
+// achieve globally optimal concrete resource alignment with respect to
+// NUMA locality.
+type HintProvider interface {
+	// GetTopologyHints returns a map of resource names to a list of possible
+	// concrete resource allocations in terms of NUMA locality hints. Each hint
+	// is optionally marked "preferred" and indicates the set of NUMA nodes
+	// involved in the hypothetical allocation. The topology manager calls
+	// this function for each hint provider, and merges the hints to produce
+	// a consensus "best" hint. The hint providers may subsequently query the
+	// topology manager to influence actual resource assignment.
+	GetTopologyHints(pod *v1.Pod, container *v1.Container) map[string][]TopologyHint
```

Comment:
Do we also need `GetPodTopologyHints`?

Reply:
> Do we also need `GetPodTopologyHints`?

yes,  if we want to allocate resources for dedicated cores pods on specific NUMA nodes, this is the only way to get available hints from QRMPlugins.

Reply:
It seems `GetPodTopologyHints` has not been defined and implemented yet. Could you also add it?

Reply:
> It seems `GetPodTopologyHints` has not been defined and implemented yet. Could you also add it?

sorry I misunderstood you question, `GetPodTopologyHints` is not implemented by QRM plugins so we don't need it. 
I add `GetPodTopologyHints` for ORM manager and it only returns nil now.